### PR TITLE
Unless there's some reason for not giving macros an items iterator.

### PR DESCRIPTION
**Pull request by transfuturist**

### CODE REVIEW

**Comment by dom96** [comment]
There is already an iterator called ``children`` which does the same thing. I wonder why it isn't called ``items``, in any case, I guess it should be deprecated.

@Araq is there a reason why the iterator is called ``children``?

**Comment by transfuturist** [comment]
Ah, deprecated is a better way to solve this.

**Comment by fowlmouth** [comment]
I named it children to make it explicit because not all nodes have children. @transfuturist can you add a check if node has children before the for loop, otherwise this causes an error when its used on a literal. (it should just not run)

File: lib/core/macros.nim [hunk 1]
```diff
@@ -708,9 +708,12 @@ proc `$`*(node: NimNode): string {.compileTime.} =
 proc ident*(name: string): NimNode {.compileTime,inline.} = newIdentNode(name)
   ## Create a new ident node from a string
 
-iterator children*(n: NimNode): NimNode {.inline.}=
-  for i in 0 .. high(n):
-    yield n[i]
+iterator items*(n: NimNode): NimNode {.inline.} =
+  if n[0] != nil:
```
Comment (dom96) [inline lib/core/macros.nim]: Might be better to check the node kind.

Comment (fowlmouth) [inline lib/core/macros.nim]: I believe the sons range is `nnkMeta .. nnkReturnToken`

Comment (transfuturist) [inline lib/core/macros.nim]: `nkType` has sons as well.

File: lib/core/macros.nim [hunk 2]
```diff
@@ -708,9 +708,12 @@ proc `$`*(node: NimNode): string {.compileTime.} =
 proc ident*(name: string): NimNode {.compileTime,inline.} = newIdentNode(name)
   ## Create a new ident node from a string
 
-iterator children*(n: NimNode): NimNode {.inline.}=
-  for i in 0 .. high(n):
-    yield n[i]
+iterator items*(n: NimNode): NimNode {.inline.} =
```
Comment (dom96) [inline lib/core/macros.nim]: Please document this iterator.

File: lib/core/macros.nim (hunk 1 above)
Comment (fowlmouth) [inline lib/core/macros.nim]: @transfuturist nnkType does not belong here, from ast.nim, this node is an atom. Are you thinking of nnkTypeDef or nnkTypeSection?
```
    nkType,               # node is used for its typ field
```

**Comment by Araq** [comment]
I don't mind 'items' but please don't deprecate 'children'. 'children' is still nice to have for documentary purposes.

**Comment by dom96** [comment]
@transfuturist ping

File: lib/core/macros.nim [hunk 3]
```diff
@@ -708,9 +708,12 @@ proc `$`*(node: NimNode): string {.compileTime.} =
...
+  if n.kind in {nnkType, nnkMetaNode .. nnkReturnToken}:
```
Comment (Araq) [inline lib/core/macros.nim]: nnkType has no children. The proper check is 'if n.kind > nnkNilLit'.

**Comment by Araq** [comment]
Somebody should rebase this.

**Comment by dom96** [comment]
On it.

**Comment by dom96** [comment]
Done.

**Pull request merged**